### PR TITLE
ddl: support session level tidb_ddl_reorg_worker_cnt and batch_size

### LINKED ISSUE #55335: Support session level `tidb_ddl_reorg_worker_cnt` and `tidb_ddl_reorg_batch_size`
## Enhancement

Currently, these two system variables are global level and cannot be modified at session level. This introduce the inconvenience when global variables are restricted to modified:

```sql
mysql> set  @@global.tidb_ddl_reorg_worker_cnt = 6;
ERROR 1227 (42000): Access denied; you need (at least one of) the SUPER or SYSTEM_VARIABLES_ADMIN privilege(s) for this operation
```

If TiDB supports session level variables, only `ALTER` privilege is needed to add index, instead of `SUPER` or `SYSTEM_VARIABLES_ADMIN`.

However, there is a slight difference between session level and global level variables.
- For global variables, the modification will also take effect on running DDL jobs. 
- For session variables, because the current session is waiting for the DDL to complete and return the result, it cannot modify this parameter online for running DDL jobs. Therefore, there is a restriction that it can only take effect for the next DDL job. In some scenarios, such as adding an index that takes several days, it will be limited.

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #55335

Problem Summary: see https://github.com/pingcap/tidb/issues/55335

### What changed and how does it work?

- Add session scope support for system variables `tidb_ddl_reorg_worker_cnt` and `tidb_ddl_reorg_batch_size`.
- For session scope variables, the value cannot be changed during DDL job execution.
- For global scope variables, the they only provide the default values of session variables.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
    Execute following SQLs on root user:
    ```
    create user 'aa'@'%' identified by '';
    create database aa;
    grant create, alter, insert on aa.* to 'aa'@'%';
    ```
    Switch to user 'aa'@'%':
    ```
    mysql> use aa;
    Database changed
    mysql> show tables;
    Empty set (0.00 sec)

    mysql> create table t (a int, b int);
    Query OK, 0 rows affected (0.07 sec)

    mysql> insert into t values (1, 1);
    Query OK, 1 row affected (0.03 sec)

    mysql> select @@global.tidb_ddl_reorg_worker_cnt;
    +------------------------------------+
    | @@global.tidb_ddl_reorg_worker_cnt |
    +------------------------------------+
    |                                  4 |
    +------------------------------------+
    1 row in set (0.00 sec)

    mysql> set  @@global.tidb_ddl_reorg_worker_cnt = 8;
    ERROR 1227 (42000): Access denied; you need (at least one of) the SUPER or SYSTEM_VARIABLES_ADMIN privilege(s) for this operation
    mysql> set @@tidb_ddl_reorg_worker_cnt = 8;
    Query OK, 0 rows affected (0.00 sec)

    mysql> alter table t add index idx(a);
    Query OK, 0 rows affected (2.20 sec)
    ```
    Check TiDB log:
    ```
    [INFO] [executor.go:4780] ["initialize reorg meta"] [category=ddl] [jobSchema=aa] [jobTable=t] [jobType="add index"] [enableDistTask=true] [enableFastReorg=true] [targetScope=] [concurrency=8] [batchSize=256]
    ```
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: pkg/ddl/backfilling_scheduler.go [hunk 1]
```diff
@@ -85,6 +85,10 @@ func newTxnBackfillScheduler(ctx context.Context, info *reorgInfo, sessPool *ses
 	if err != nil {
 		return nil, err
 	}
+	workerCnt := info.ReorgMeta.Concurrency
+	if workerCnt == 0 {
+		workerCnt = int(variable.GetDDLReorgWorkerCounter())
+	}
```
[the remark this answers is not in the document]
Comment (tangenta) [reply pkg/ddl/backfilling_scheduler.go]: https://github.com/pingcap/tidb/pull/55334/commits/9e9594038a4e7999433994286f8a39854f758f49

File: pkg/parser/model/reorg.go [hunk 2]
```diff
@@ -30,6 +30,8 @@ type DDLReorgMeta struct {
 	ReorgTp           ReorgType                        `json:"reorg_tp"`
 	IsFastReorg       bool                             `json:"is_fast_reorg"`
 	IsDistReorg       bool                             `json:"is_dist_reorg"`
+	Concurrency       int                              `json:"concurrency"`
```
[the remark this answers is not in the document]
Comment (tangenta) [reply pkg/parser/model/reorg.go]: https://github.com/pingcap/tidb/pull/55334/commits/9e9594038a4e7999433994286f8a39854f758f49

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**